pkg/controller/certificate-shim/checks.go

```diff
@@ -19,28 +19,32 @@ package controller
 import (
 	"fmt"
 
-	networkingv1beta1 "k8s.io/api/networking/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	v1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 )
 
-func (c *controller) ingressesForCertificate(crt *v1.Certificate) ([]*networkingv1beta1.Ingress, error) {
-	ings, err := c.ingressLister.List(labels.NewSelector())
+func (c *controller) objectsForCertificate(crt *v1.Certificate) ([]runtime.Object, error) {
+	objs, err := c.objectLister.List(labels.NewSelector())
```

Comment:
question: from afar, it reads like we are listing "all" objects of all kinds. In reality, this lister only lists ingresses and gateway API kinds. Would it make sense to rename this field to `ingressAndSimilarLister` or something like that?

Reply:
The function should probably be called ownersOfCertificate - it returns the controller in the given certificate's OwnerRef